boards/nucleo-f4xx: Added second and third SPI definition

phectori opened the pull request:
Enables the second SPI interface of the nucleo-f401 board. The second interface is only available on the Morpho header, it is not available on the Arduino compatible header. 

Tested with a logic analyzer and the enc28j60 ethernet module.

phectori commented:
Ping?

aabadie commented:
There's a rework of the SPI API in progress with #4780 and we are waiting for it before going forward here. Just be patient ;)

aabadie commented:
@PeterKietzmann, Unfortunately, yesterday I burnt the nucleo-f401 I have here and thus can no longer test this one...

phectori commented:
#4780 is merged, so I have changed the SPI definitions to conform to the new SPI API. Also, I have added SPI3 to the struct.

I have added the same definitions to the nucleo-f446 board peripherals. Haven't had the time to test those yet.

aabadie commented on boards/nucleo-f446/include/periph_conf.h:
```diff
@@ -200,6 +200,26 @@ static const spi_conf_t spi_config[] = {
+    {
+        .dev      = SPI3,
+        .mosi_pin = GPIO_PIN(PORT_C, 12),
+        .miso_pin = GPIO_PIN(PORT_C, 11),
```
PC10 and PC11 are already used for the second UART. It may not work in some cases. Do you really need the third SPI ?

phectori replied:
I have designed a small adapter that connects the Nucleo boards to the mrf24j40ma.

https://github.com/phectori/mrf24j40ma-nucleo

I have used SPI3 because it is close to some power and ground pins, this way I could make the PCB smaller.

aabadie replied:
on the f401 it's fine but I'm wondering if it still works with the f446. Have you tried on this one ?
Maybe it will if you don't initialize the second UART, I just want to be sure.

phectori replied:
Ok, I don't have a f446 at hand right now. I can check tomorrow evening (UTC+1).

phectori replied:
So, I have checked SPI3 on the nucleo-f446 with a logic analyzer. It seems to work fine.

aabadie replied:
Thanks ! Have you also tried the second UART ?

phectori replied:
Now I have :p. The second UART also works with the logic analyzer.

aabadie commented:
Murdock2 is reporting an error that is not related. @kaspar030, any idea ?

aabadie commented:
ping @kaspar030. Should we reschedule a build on Murdock2  or is it ok for merge ?

phectori commented:
I just rebased my branch on current master. I don't understand why github lists all these other commits now. The files changed seem to be correct though.

aabadie commented:
Yes, that's strange. Did you rebase or merge ? or maybe git pull ?
The good way of rebasing is either: git rebase origin/master or git pull --rebase

Can you try again ?

aabadie commented:
and `origin` should point to the upstream remote `git@github.com:RIOT-OS/RIOT.git`

phectori commented:
What I did was:

    git rebase master

using `origin/master` seemed to do the trick.

aabadie commented:
ah yes. Thanks !

phectori commented:
I just wanted to trigger a new Murdock(2) build :S. It seems to succeed now :).

aabadie commented:
All green, go !
